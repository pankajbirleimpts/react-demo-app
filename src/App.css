body {
  background-color: #f7f7f7;
  margin-top: -16px !important;
}

.header-name-section {
  background-color: #23d0e8;
  color: #fff;
  padding: 5px;
  min-height: 80px;
}

p.red {
  color: red;
}

.user-details {
  text-align: right;
}

.user-details p {
  margin: 2px 0;
  font-size: 16px;
}

.user-details .user-role {
  font-size: 14px;
}

.user-details a {
  color: #fff;
}

.user-details a:hover {
  color: #fff;
}

.web-logo {
  font-style: italic;
  font-size: 22px;
}

.menu-container {
  padding-top: 0 !important;
}

.routing-container {
  min-height: 550px;
}

.alert-warning {
  text-align: center;
}

.label-login-as-admin {
  margin-left: 13px;
  color: rgba(0, 0, 0, .87);
  font-size: .92857143em;
  font-weight: 700;
  text-transform: none;
  text-align: center;
  display: block;
  float: right;
  margin-right: 72%;;
}