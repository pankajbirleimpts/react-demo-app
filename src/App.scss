$redColor: #d60010;
p.red {
  color: $redColor;
  font-size: 12px;
}
