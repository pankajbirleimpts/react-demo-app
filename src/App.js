import React from "react";
import "./App.css";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { ToastContainer } from "react-toastify";
import { BrowserRouter as Router } from 'react-router-dom';
import rootReducer from "./reducers/RootReducer";
import Routing from "./components";
import "react-toastify/dist/ReactToastify.min.css";
import 'react-confirm-alert/src/react-confirm-alert.css'; // Import css

const store = createStore(rootReducer, applyMiddleware(thunk));
function App() {
  return (
    <Router>
      <Provider store={store}>
        <ToastContainer autoClose={8000} />
        <Routing />
      </Provider>
    </Router>
  );
}

export default App;
