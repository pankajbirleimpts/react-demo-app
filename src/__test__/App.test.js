import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { storeFactory } from './test/TestUtils';

const setup = (props) => {
  const store = storeFactory(props);
  const wrapper = shallow(<App store={store} />)
    .dive()
    .dive();
  return wrapper;
};

describe('App component', () => {
  test('Snapshot of App component', () => {
    const props = {
      user: {
        isAuthenticated: false,
      },
    };
    const wrapper = setup(props);
    expect(wrapper).toMatchSnapshot();
  });

});
