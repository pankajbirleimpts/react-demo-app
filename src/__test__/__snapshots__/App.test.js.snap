// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot of App component 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ToastContainer
      autoClose={8000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <Connect(UnConnectedRouting) />
  </Provider>
</ContextProvider>
`;
